src/argparse.cpp

```diff
@@ -0,0 +1,67 @@
+#include "argparse.h"
+
+#include <iostream>
+
+using std::string;
+
+/**
+ * @brief try to parse the arguments
+ * @param input
+ * @param output
+ * @return return whether there has been an error (true = error, false = no error)
+ */
+bool ArgParse::parseOrExit(Input& input, Output& output)
```

Comment:
The name is a bit confusing IMHO, especially if you see it in use.
if(parseOrExit) is not obvious just by looking at the method name, only after consulting the doc string.
Could be a bit more obvious, e.g. inputParsed, or another name that reflects some kind of success/failure?
What do you think?

Reply:
I've renamed it to `parsingFails()`, because then something like this reads well:
```C++
if (ap.parsingFails(input, output)) {
    return ap.exitCode();
}
```
(For `inputParsed()` I think its ambigious whether `true` means "error" or "success")

Reply:
That is definitely better!